spec.html

```diff
@@ -32,4 +32,94 @@
   }
 </style>
 
-TODO
+<emu-clause id="sec-native-error-types-used-in-this-standard-aggregatenerror">
+  <h1>AggregateError</h1>
+  <p>TODO implement</p>
+</emu-clause>
+
+<emu-intro id="intro">
+  <h1>Introduction</h1>
+  <p>`Promise.any()` accepts an array of promises and returns a promise that is fulfilled by the first given promise to be fulfilled, or rejected with an array of rejection reasons if all of the given promises are rejected.</p>
+</emu-intro>
+
+<emu-clause id="sec-promise.allSettled">
+  <h1>Promise.any ( _iterable_ )</h1>
+  <p>The `any` function returns a promise that is fulfilled by the first given promise to be fulfilled, or rejected with an array of rejection reasons if all of the given promises are rejected. It resolves all elements of the passed iterable to promises as it runs this algorithm.</p>
+  <emu-alg>
+    1. Let _C_ be the *this* value.
+    1. If Type(_C_) is not Object, throw a *TypeError* exception.
+    1. Let _promiseCapability_ be ? NewPromiseCapability(_C_).
+    1. Let _iteratorRecord_ be GetIterator(_iterable_).
+    1. IfAbruptRejectPromise(_iteratorRecord_, _promiseCapability_).
+    1. Let _result_ be PerformPromiseAny(_iteratorRecord_, _C_, _promiseCapability_).
+    1. If _result_ is an abrupt completion, then
+      1. If _iteratorRecord_.[[Done]] is *false*, set _result_ to IteratorClose(_iteratorRecord_, _result_).
+      1. IfAbruptRejectPromise(_result_, _promiseCapability_).
+    1. Return Completion(_result_).
+  </emu-alg>
+  <p>This function is the <dfn>%Promise_any%</dfn> intrinsic object.</p> <!-- not sure about this -->
+  <emu-note>
+    <p>The `any` function requires its *this* value to be a constructor function that supports the parameter conventions of the `Promise` constructor.</p>
+  </emu-note>
+
+  <emu-clause id="sec-performpromiseany" aoid="PerformPromiseAny">
+    <h1>Runtime Semantics: PerformPromiseAny ( _iteratorRecord_, _constructor_, _resultCapability_ )</h1>
+    <p>When the PerformPromiseAny abstract operation is called with arguments _iteratorRecord_, _constructor_, and _resultCapability_, the following steps are taken:</p>
+    <emu-alg>
+      1. Assert: ! IsConstructor(_constructor_) is *true*.
+      1. Assert: _resultCapability_ is a PromiseCapability Record.
+      1. Let _values_ be a new empty List.
+      1. Let _remainingElementsCount_ be a new Record { [[Value]]: 1 }.
```

Comment:
It seems weird to have a record here and pass it around. Is keeping a count like this the only solution?

Reply:
Looking at existing [ECMAScript Specification Types](https://tc39.github.io/ecma262/#sec-ecmascript-specification-types), `Record` is most suitable type here. It's used in other `Promise` namespace functions as well.

Reply:
oh sure, i wasn't suggesting a JS object or anything :-) more that, is passing around a counter the only viable approach?

Reply:
Oh I got it. Not sure to be honest 🤔 I need to do some more research to answer this question 🕵️‍♀️